uaclient/api/tests/test_u_pro_attach_auto_full_auto_attach_v1.py

```diff
@@ -0,0 +1,176 @@
+import mock
+import pytest
+
+from uaclient.api import exceptions
+from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import (
+    FullAutoAttachOptions,
+    _full_auto_attach,
+    _is_incompatible_services_present,
+)
+from uaclient.entitlements.entitlement_status import (
+    CanEnableFailure,
+    CanEnableFailureReason,
+)
+from uaclient.messages import NamedMessage
+
+M_API = "uaclient.api.u.pro."
+
+
+class TestFullAutoAttachV1:
+    @mock.patch(
+        "uaclient.actions.enable_entitlement_by_name",
+        return_value=(True, None),
+    )
+    def test_error_when_beta_in_enable_list(
+        self,
+        enable_ent_by_name,
+        FakeConfig,
+    ):
+        cfg = FakeConfig(root_mode=True)
+        opts = {"enable": ["esm-infra", "realtime-kernel"]}
+        options = FullAutoAttachOptions.from_dict(opts)
```

Comment:
Sorry I didn't look closely at these tests before. There's no need to create all these Options objects from dicts. And there is a reason to create them directly: mypy will help us pass the correct args.

I think all of the tests in this file should construct the Options object directly like so
```suggestion
        options = FullAutoAttachOptions(enable=["esm-infra", "realtime-kernel"])
```